package image;

message ClassifyRequest {
  enum RequestType {
    CLASSIFY = 0;
    SIMILARITY = 1;
    SEARCH = 2;
    CLASSIFY_PAIPAI = 3;  //拍拍分类器
    SEMANTIC = 4;
    CLASSIFY_ALEX = 5;
  }
  // file_name 与 image_url 任意选择一个填入.如果选择image_url,则触发下载逻辑
  optional bytes file_name = 1;
  optional int32 top_n_result = 2 [default = 5];
  optional bytes compare_file_name = 3;
  optional RequestType request_type = 4 [default = CLASSIFY];
  optional float min_sim_thres = 5 [default = 0.1];
  optional float max_sim_thres = 6 [default = 1.0];
  optional int32 feature_layer = 7 [default = 20];
  optional bytes image_url = 8;
}


message ClassifyResult {
  optional bytes category_name = 1;
  optional int32 category_id = 2;
  optional float category_weight = 3;
}
message SearchResult {
  optional bytes search_file_name = 1;
  optional float search_similarity = 2;
}
message SearchFeature {
  optional bytes feature_token = 1;
  optional float feature_weight = 2;
}
message ClassifyResponse {
  repeated ClassifyResult rsp_res = 1;
  optional float similarity = 2;
  repeated SearchResult search_res = 3;
  repeated SearchFeature search_feature = 4;
  // 待分类的文件名.如果request.filename有值,则classify_filename=request.filename
  // 如果没有值,classify_filename则是利用image_url下载后的图片的文件路径
  optional bytes classify_filename = 5;
}
